33-lifecycle-lab_v16/src/03-custominput.jsx

```diff
@@ -1,10 +1,80 @@
-/**
- * Implment an input component that shows 4
- * input boxes writing user input in the currently
- * focused panel
- *
- * See lab description on the webpage for a live example
- * how this should work.
- */
-
-
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class SquareFocus extends React.Component{
+
+    constructor(props) {
+        super(props);
+        this.state = {texts : []};

```

Comment:
I don't think you gained much by saving an array of objects here (vs. an array of strings). Code would have been simpler if texts was: `['x', 'y', '1', '2']` rather than your version: `[{ txt: 'x'}, { txt: 'y' }, { txt: '1' }, { txt: '2' }]`